### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by brim-borium**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,15 +20,16 @@ track of the issue.
 
 ## What do I do before creating a pull request
 
-1. Fork the repository and create your branch from `master`.
-1. Install all dependencies (`flutter packages get` or `pub get`).
-1. Squash your commits and ensure you have a meaningful commit message.
+1. Fork the repository and create your branch from `master`, use feature/ , bug/ or task/ to differentiate your branch
```
Comment (fotiDim) [inline CONTRIBUTING.md:23]:
```suggestion
1. Fork the repository and branch out of `master`, prefix your branch's name with `feature/`, `bug/` or `task/` to indicate the scope of the PR
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,15 +20,16 @@ track of the issue.
...
+1. Install all dependencies (`flutter packages get` or `pub get`)
+1. Squash your commits and ensure you have a meaningful commit message, use [imperative mood](https://chris.beams.io/posts/git-commit/#imperative)
 1. If you’ve fixed a bug or added code that should be tested, add tests!
 1. If you've changed the public API, make sure to update/add documentation (for now that is the [Readme](README.md))
-1. Format your code (`dartfmt -w .`).
-1. Analyze your code (`dartanalyzer --fatal-infos --fatal-warnings .`).
-1. Create the Pull Request.
-1. Verify that all status checks are passing.
+1. If you've made breaking changes give us a heads up in the pull request
```
Comment (fotiDim) [inline CONTRIBUTING.md:28]:
```suggestion
1. If you've made breaking changes give us a heads up in the pull request. Try to provide a compatibility path for the deprecated APIs and if needed provide a small migration guide in the [Readme](README.md).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -20,15 +20,16 @@ track of the issue.
 
 ## What do I do before creating a pull request
 
-1. Fork the repository and create your branch from `master`.
-1. Install all dependencies (`flutter packages get` or `pub get`).
-1. Squash your commits and ensure you have a meaningful commit message.
+1. Fork the repository and branch out of `master`, prefixing your branch's name with `feature/`, `bug/` or `task/` to indicate the scope of the PR.
+1. Install all dependencies (`flutter packages get` or `pub get`)
+1. Squash your commits and ensure you have a meaningful commit message, use [imperative mood](https://chris.beams.io/posts/git-commit/#imperative)
```
Comment (fotiDim) [inline CONTRIBUTING.md:25]: We can also squash through GitHub when we merge the pull request. In fact we can even configure it to be mandatory.
```suggestion
1. Ensure you have a meaningful PR name using the [imperative mood](https://chris.beams.io/posts/git-commit/#imperative) as your commits will be squashed upon merge and the PR's name will be used as the merge commit's message.
```

Comment (brim-borium) [reply CONTRIBUTING.md:25]: Sound good, can I configure that in the github settings?

Comment (brim-borium) [reply CONTRIBUTING.md:25]: Found it Squash is mandatory now

**Comment by brim-borium** [comment]
> One last point before merging. What do you think?

Valid point, changed the repo so that squash is mandatory now

**Pull request merged**